packages/core/src/routes/interaction/index.test.ts

```diff
@@ -89,171 +102,191 @@ mockEsmDefault(
-  () => jest.fn(koaSessionSignInExperienceGuard)
-);
-
 const {
   default: interactionRoutes,
   verificationPrefix,
   interactionPrefix,
 } = await import('./index.js');
 
 describe('session -> interactionRoutes', () => {
+  const baseProviderMock = {
+    params: {},
+    jti: 'jti',
+    client_id: demoAppApplicationId,
+  };
   const sessionRequest = createRequester({
     anonymousRoutes: interactionRoutes,
-    provider: createMockProvider(
-      jest.fn().mockResolvedValue({ params: {}, jti: 'jti', client_id: demoAppApplicationId })
-    ),
+    provider: createMockProvider(jest.fn().mockResolvedValue(baseProviderMock)),
   });
 
   afterEach(() => {
     jest.clearAllMocks();
   });
 
   describe('PUT /interaction', () => {
-    const path = interactionPrefix;
+    const path = `${interactionPrefix}`;
 
-    it('sign-in event should call methods properly', async () => {
+    it('should call validations properly', async () => {
       const body = {
         event: Event.SignIn,
-        identifier: {
-          username: 'username',
-          password: 'password',
-        },
+        identifier: { email: 'email@logto.io', password: 'password' },
+        profile: { phone: '1234567890' },
       };
       const response = await sessionRequest.put(path).send(body);
-      expect(koaInteractionBodyGuardSpy).toBeCalled();
-      expect(koaSessionSignInExperienceGuardSpy).toBeCalled();
-      expect(verifyIdentifier).toBeCalled();
-      expect(verifyProfile).toBeCalled();
-      expect(validateMandatoryUserProfile).toBeCalled();
-      expect(submitInteraction).toBeCalled();
-      expect(response.status).toEqual(200);
-      expect(response.body).toEqual({ redirectUri: 'logto.io' });
+      expect(getSignInExperience).toBeCalled();
+      expect(verifySignInModeSettings).toBeCalled();
+      expect(verifyIdentifierSettings).toBeCalled();
+      expect(verifyProfileSettings).toBeCalled();
+      expect(verifyIdentifierPayload).toBeCalled();
+      expect(storeInteractionResult).toBeCalled();
+      expect(response.status).toEqual(204);
     });
+  });
 
-    it('forgot password event should not call UserProfile validation', async () => {
-      const body = {
-        event: Event.ForgotPassword,
-        identifier: {
-          email: 'email@logto.io',
-          passcode: 'passcode',
-        },
-        profile: {
-          password: 'password',
-        },
-      };
-
-      const response = await sessionRequest.put(path).send(body);
-
-      expect(verifyIdentifier).toBeCalled();
-      expect(verifyProfile).toBeCalled();
-      expect(validateMandatoryUserProfile).not.toBeCalled();
-      expect(submitInteraction).toBeCalled();
-      expect(response.status).toEqual(200);
+  describe('DELETE /interaction', () => {
+    it('should call assignInteraction Result properly', async () => {
+      await sessionRequest.delete(`${interactionPrefix}`);
+      expect(assignInteractionResults).toBeCalled();
```

Comment:
nit: Does calling `assignIteractionResults` enough to prove the delete request is called/successful?

Reply:
yes, delete the interaction, simply assign an empty interaction to the session.  This is an migrated existing API.